docs/collector.smbclient.md

```diff
@@ -0,0 +1,37 @@
+# smb collector
+
+The smbclient collector collects metrics from MS SmbClient hosts through perflib
+=======
+
+
+|||
+-|-
+Metric name prefix  | `smbclient`
+Classes 			| [Win32_PerfRawData_SMB](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-smb/)<br/> 
+Enabled by default? | No
+
+## Flags
+
+### `--collectors.smbclient.list`
+Lists the Perflib Objects that are queried for data along with the perlfib object id
+
+### `--collectors.smbclient.enabled`
+Comma-separated list of collectors to use, for example: `--collectors.smbclient.enabled=ServerShares`. Matching is case-sensitive. Depending on the smb protocol version not all performance counters may be available. Use `--collectors.smbclient.list` to obtain a list of supported collectors.
+
+## Metrics
+Name          | Description
+--------------|---------------
+`windows_smbclient_client_shares_avg_sec_per_read` | The average latency between the time a read request is sent and when its response is received.
+`windows_smbclient_client_shares_avg_sec_per_write` | The average latency between the time a write request is sent and when its response is received.
+
+### Example metric
+windows_smbclient_read_bytes_total - The bytes read on this share.
+windows_smbclient_write_bytes_total - The bytes written on this share.
```

Comment:
Please provider examples from the `/metrics` endpoint, including all metrics and labels

Reply:
Is this what you want? These are all the `windows_smbclient` metrics produced on a scrape.
```
# HELP windows_smbclient_current_data_queues The point in time number of requests outstanding on this share.
# TYPE windows_smbclient_current_data_queues counter
windows_smbclient_current_data_queues{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_current_data_queues{server="localhost",share="c$"} 0
# HELP windows_smbclient_data_bytes_total The bytes read or written on this share.
# TYPE windows_smbclient_data_bytes_total counter
windows_smbclient_data_bytes_total{server="10.57.120.60",share="genshare0"} 2.19776e+07
windows_smbclient_data_bytes_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_metadata_requests_total The metadata requests on this share.
# TYPE windows_smbclient_metadata_requests_total counter
windows_smbclient_metadata_requests_total{server="10.57.120.60",share="genshare0"} 30
windows_smbclient_metadata_requests_total{server="localhost",share="c$"} 167
# HELP windows_smbclient_read_bytes_total The bytes read on this share.
# TYPE windows_smbclient_read_bytes_total counter
windows_smbclient_read_bytes_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_read_bytes_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_read_bytes_via_smbdirect_total The bytes read from this share via RDMA direct placement.
# TYPE windows_smbclient_read_bytes_via_smbdirect_total counter
windows_smbclient_read_bytes_via_smbdirect_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_read_bytes_via_smbdirect_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_read_requests_total The read requests on this share.
# TYPE windows_smbclient_read_requests_total counter
windows_smbclient_read_requests_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_read_requests_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_read_requests_via_smbdirect_total The read requests on this share via RDMA direct placement.
# TYPE windows_smbclient_read_requests_via_smbdirect_total counter
windows_smbclient_read_requests_via_smbdirect_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_read_requests_via_smbdirect_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_read_seconds_total Seconds waiting for read requests on this share.
# TYPE windows_smbclient_read_seconds_total counter
windows_smbclient_read_seconds_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_read_seconds_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_reads_queued The number of read requests queued on this share.
# TYPE windows_smbclient_reads_queued counter
windows_smbclient_reads_queued{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_reads_queued{server="localhost",share="c$"} 0
# HELP windows_smbclient_request_seconds_total Seconds waiting for requests on this share.
# TYPE windows_smbclient_request_seconds_total counter
windows_smbclient_request_seconds_total{server="10.57.120.60",share="genshare0"} 32.2819863
windows_smbclient_request_seconds_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_requests_queued_total The number of requests queued on this share.
# TYPE windows_smbclient_requests_queued_total counter
windows_smbclient_requests_queued_total{server="10.57.120.60",share="genshare0"} 3.22819863e+08
windows_smbclient_requests_queued_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_requests_total The requests on this share.
# TYPE windows_smbclient_requests_total counter
windows_smbclient_requests_total{server="10.57.120.60",share="genshare0"} 21
windows_smbclient_requests_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_stalls_total The number of requests delayed based on insufficient credits on this share.
# TYPE windows_smbclient_stalls_total counter
windows_smbclient_stalls_total{server="10.57.120.60",share="genshare0"} 3
windows_smbclient_stalls_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_turbo_io_reads_total The read requests that go through Turbo I/O
# TYPE windows_smbclient_turbo_io_reads_total counter
windows_smbclient_turbo_io_reads_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_turbo_io_reads_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_turbo_io_writes_total The write requests that go through Turbo I/O
# TYPE windows_smbclient_turbo_io_writes_total counter
windows_smbclient_turbo_io_writes_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_turbo_io_writes_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_write_bytes_total The bytes written on this share.
# TYPE windows_smbclient_write_bytes_total counter
windows_smbclient_write_bytes_total{server="10.57.120.60",share="genshare0"} 2.19776e+07
windows_smbclient_write_bytes_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_write_bytes_via_smbdirect_total The written bytes to this share via RDMA direct placement.
# TYPE windows_smbclient_write_bytes_via_smbdirect_total counter
windows_smbclient_write_bytes_via_smbdirect_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_write_bytes_via_smbdirect_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_write_requests_total The write requests on this share.
# TYPE windows_smbclient_write_requests_total counter
windows_smbclient_write_requests_total{server="10.57.120.60",share="genshare0"} 21
windows_smbclient_write_requests_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_write_requests_via_smbdirect_total The write requests to this share via RDMA direct placement.
# TYPE windows_smbclient_write_requests_via_smbdirect_total counter
windows_smbclient_write_requests_via_smbdirect_total{server="10.57.120.60",share="genshare0"} 0
windows_smbclient_write_requests_via_smbdirect_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_write_seconds_total Seconds waiting for write requests on this share.
# TYPE windows_smbclient_write_seconds_total counter
windows_smbclient_write_seconds_total{server="10.57.120.60",share="genshare0"} 32.2819863
windows_smbclient_write_seconds_total{server="localhost",share="c$"} 0
# HELP windows_smbclient_writes_queued The number of write requests queued on this share.
# TYPE windows_smbclient_writes_queued counter
windows_smbclient_writes_queued{server="10.57.120.60",share="genshare0"} 3.22819863e+08
windows_smbclient_writes_queued{server="localhost",share="c$"} 0
```

Reply:
Yes!